x-pack/platform/plugins/private/snapshot_restore/common/lib/time_serialization.ts

```diff
@@ -5,11 +5,16 @@
  * 2.0.
  */
 
+import type { Duration } from '@elastic/elasticsearch/lib/api/types';
 import { TIME_UNITS } from '../constants';
 
-export const deserializeTime = (time: string) => {
+export const deserializeTime = (time: Duration) => {
   const timeUnits = Object.values(TIME_UNITS);
 
+  if (typeof time !== 'string') {
```

Comment:
Whats going on here? Seems like a strange manual type check after changing `time` away from `string`

Reply:
`Duration` can be `string | -1 | 0`.

Reply:
So if duration is 0 or -1 its pretty much empty, right? Would be nice to just have a code comment

Reply:
Good point! I added a comment in [`fe6f67f` (#208776)](https://github.com/elastic/kibana/pull/208776/commits/fe6f67feb64c7af58aeb01c9ef6ddf8e7744380c)